crates/medea-macro/src/js_caused.rs

```diff
@@ -0,0 +1,114 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use synstructure::{BindStyle, Structure};
+
+/// Generates the actual code for `#[derive(JsCaused)]` macro.
+///
+/// # Algorithm
+///
+/// 1. Generate body for trait method `name()` as `enum` variant name as is.
+/// 2. Generate body for trait method `js_cause()`:
+///     - if `enum` variant contains associated error, returns this error;
+///     - if `enum` variant contains `JsCaused`, invoke trait method;
+///     - otherwise returns `None`.
+/// 3. Generate implementation `JsCaused` trait for this `enum` with generated
+/// methods from step 1 and 2.
+#[allow(clippy::needless_pass_by_value)]
+pub fn derive(mut s: Structure) -> TokenStream {
+    let error_type = error_type(&s).expect("not found js_error attribute");
```

Comment:
Don't make it mandatory. This explicit `#[js_error(JsError)]` everywhere makes a lot of confusion. Make the reasonable `JsError` default value.